config/appstructure.go

```diff
@@ -0,0 +1,929 @@
+	appStructureScenario := make([]scenario.Action, 0, len(actions))
+	for _, act := range actions {
+		info, subActions := act.AppStructureAction()
+		if info != nil {
+			if info.Include {
+				appStructureScenario = append(appStructureScenario, act)
+			}
+			if info.IsAppAction {
+				appStructureScenario = append(appStructureScenario, scenario.Action{
+					ActionCore: scenario.ActionCore{
+						Type:  "getappstructure",
+						Label: "Get app structure",
+					},
+					Settings: &getAppStructureSettings{
+						IncludeRaw: includeRaw,
+						Summary:    summary,
+					},
+				})
+
+			}
+			if len(subActions) > 0 {
+				appStructureScenario = append(appStructureScenario, evaluateActionList(subActions, includeRaw, summary)...)
+			}
+		}
+	}
+	return appStructureScenario
+}
+
+// GetAppStructures for all apps in scenario
+func (cfg *Config) GetAppStructures(ctx context.Context, includeRaw bool) error {
+	// find all auth and actions
+	appStructureScenario := cfg.getAppStructureScenario(includeRaw, cfg.Settings.LogSettings.getSummaryType())
+	if len(appStructureScenario) < 1 {
+		return AppStructureNoScenarioActionsError{}
+	}
+
+	// Replace scheduler with 1 iteration 1 user simple scheduler
+	cfg.Scheduler = &scheduler.SimpleScheduler{
+		Scheduler: scheduler.Scheduler{
+			SchedType: scheduler.SchedSimple,
+		},
+		Settings: scheduler.SimpleSchedSettings{
+			ExecutionTime:   -1,
+			Iterations:      1,
+			ConcurrentUsers: 1,
+			RampupDelay:     1.0,
+		},
+	}
+
+	if err := cfg.Scheduler.Validate(); err != nil {
+		return errors.WithStack(err)
+	}
+
+	// Setup outputs folder
+	outputsDir, err := setupOutputs(cfg.Settings.OutputsSettings)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	logSettings := cfg.Settings.LogSettings
+
+	fileName := cfg.Settings.LogSettings.FileName.String()
+	ext := filepath.Ext(fileName)
+	appStructureLogPath := fmt.Sprintf("%s-appstructure%s", strings.TrimSuffix(fileName, ext), ext)
+	stmpl, err := session.NewSyncedTemplate(appStructureLogPath)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	logSettings.FileName = *stmpl
+
+	log, err := setupLogging(ctx, logSettings, nil, nil)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	logEntry := log.NewLogEntry()
+	logEntry.Log(logger.DebugLevel, fmt.Sprintf("outputs folder: %s", outputsDir))
+
+	timeout := time.Duration(cfg.Settings.Timeout) * time.Second
+	if err := cfg.Scheduler.Execute(ctx, log, timeout, appStructureScenario, outputsDir, cfg.LoginSettings, &cfg.ConnectionSettings); err != nil {
+		return errors.WithStack(err)
+	}
+
+	return nil
+}
+
+// Validate implements ActionSettings interface
+func (settings *getAppStructureSettings) Validate() error {
+	return nil
+}
+
+// Execute implements ActionSettings interface
+func (settings *getAppStructureSettings) Execute(sessionState *session.State, actionState *action.State, connectionSettings *connection.ConnectionSettings, label string, reset func()) {
```

Comment:
I feel like the action deserves its own file

Reply:
was contemplating it, but thought it looked weird to have a "loose" action in the `config` package and don't want to have it as a registered action.

Reply:
but yea, the appstructure.go file is a bit big right now.

Reply:
separated the actual action to getappstructure.go, didn't find much else that felt ok to separate.